src/Loader.php

```diff
@@ -60,12 +72,20 @@ private function __construct( $services ) {
 	 * Load required files.
 	 */
 	private function load_files() {
+		// Composer install might have happened inside the master-theme directory, instead of the app root.
+		// Probably there's a better way to handle this, but for now let's try load both.
+		if ( file_exists( __DIR__ . '/../vendor/autoload.php' ) ) {
+			require_once __DIR__ . '/../vendor/autoload.php';
+		} else {
+			require_once __DIR__ . '/../../../../../vendor/autoload.php';
```

Comment:
don't know if there's a better way yet 🤷‍♂️ but here's 2 versions with less dots / less based on current dir
```php
// equivalent to __DIR__ . '/../vendor/autoload.php';
dirname(__DIR__, 1) . '/vendor/autoload.php';
\get_template_directory() . '/vendor/autoload.php';

// equivalent to __DIR__ . '/../../../../../vendor/autoload.php';
dirname(__DIR__, 5) . '/vendor/autoload.php';
dirname(\ABSPATH) . '/vendor/autoload.php';
```
for whatever reason, [`get_home_path()`](https://developer.wordpress.org/reference/functions/get_home_path/) is not available there :|

Reply:
My +1 for moving a autoloading statement in functions.php and dropping a `load_files()` method from Loader class.

Reply:
It would be nice to be able to remove the custom function already, one concern that I have is that this changes the loading order, which might cause some unexpected surprises. Like you encountered with `ImageCompression`. There might be more of these cases so we would need to check all usages of WP files if we would need to add the `require_once` calls to our files.

Reply:
> I've tried a different approach for the autoload, the diff is in this gist https://gist.github.com/lithrel/78e1a7824292d162bb4d01a6b0c2afd8
Yes that's definitely better. I included the changes from the gist but just moved it in a separate file. For the `ImageCompression` class it actually makes sense to require the files as the class doesn't work without them anyway. :+1: